Source/Models/FORMGroup.m

```diff
@@ -118,4 +118,25 @@ - (NSString *)description
             self.formID, self.title, self.position, self.shouldValidate ? @"YES" : @"NO", fields];
 }
 
+- (void)removeSection:(FORMSection *)section
+{
+    __block BOOL found = NO;
+    __block NSInteger index = 0;
+
+    [self.sections enumerateObjectsUsingBlock:^(FORMSection *currentSection, NSUInteger idx, BOOL *stop) {
+        if (found) {
+            currentSection.position = @([currentSection.position integerValue] - 1);
+        }
+
+        if ([currentSection.sectionID isEqualToString:section.sectionID]) {
+            index = idx;
+            found = YES;
+        }
+    }];
+
+    if (!found) index = [self.sections count];
```

Comment:
@callumboddy you're right, this isn't the expected behavior, that's what I'm trying to accomplish with this PR. We had this as a workaround because we haven't been updating the position of the sections when a section was inserted or removed.

I updated the PR description on what I'm working on :)